src/Analyzers/CSharp/Analyzers/NewLines/ConditionalExpressionPlacement/ConditionalExpressionPlacementDiagnosticAnalyzer.cs

```diff
@@ -29,38 +29,15 @@ public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
             => DiagnosticAnalyzerCategory.SyntaxTreeWithoutSemanticsAnalysis;
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterCompilationStartAction(context =>
-                context.RegisterSyntaxTreeAction(treeContext => AnalyzeTree(treeContext, context.Compilation.Options)));
+            => context.RegisterSyntaxNodeAction(ProcessConditionalExpression, SyntaxKind.ConditionalExpression);
 
-        private void AnalyzeTree(SyntaxTreeAnalysisContext context, CompilationOptions compilationOptions)
+        private void ProcessConditionalExpression(SyntaxNodeAnalysisContext context)
         {
             var option = context.GetCSharpAnalyzerOptions().AllowBlankLineAfterTokenInConditionalExpression;
-            if (option.Value || ShouldSkipAnalysis(context, compilationOptions, option.Notification))
+            if (option.Value || ShouldSkipAnalysis(context, option.Notification))
```

Comment:
@mavasani it would be really good if we could register a start action for syntax node actions that would allow doing these skip-analysis tests when hitting the file the first time.